src/Xamarin.Android.Tools.Bytecode/ClassPath.cs

```diff
@@ -240,6 +240,15 @@ JavaDocletType GetDocletType (string path)
 				kind = JavaDocletType.Java7;
 			else if (rawHTML.Contains ("Generated by javadoc (1.8"))
 				kind = JavaDocletType.Java8;
+
+			// Check to see if it's an api.xml formatted doc
+			buf = new char[buf.Length];
+			using (var reader = File.OpenText(path))
+				reader.ReadBlock(buf, 0, buf.Length);
```

Comment:
You shouldn't ignore the return value. You should keep it, for the subsequent `new string()` invocation.

This allows you to *reuse* the existing `buf` instance w/o creating a new one.